var/spack/repos/builtin/packages/amrex/package.py

```diff
@@ -113,6 +114,7 @@ class Amrex(CMakePackage, CudaPackage):
     conflicts('cuda_arch=21', when='+cuda', msg='AMReX only supports compute capabilities >= 3.5')
     conflicts('cuda_arch=30', when='+cuda', msg='AMReX only supports compute capabilities >= 3.5')
     conflicts('cuda_arch=32', when='+cuda', msg='AMReX only supports compute capabilities >= 3.5')
+    conflicts('+rocm', when='@:20.11', msg='AMReX Hypre support needs AMReX newer than version 20.11')
```

Comment:
Should this read:

`msg='AMReX HIP support needs AMReX newer than version 20.11'`

(replace `Hypre` with `HIP`)

?

Other than that this looks excellent.

It might be useful to add a `conflicts` statement preventing `+cuda+rocm` like:

```
conflicts('+cuda', when='+hip', msg='CUDA and HIP support are exclusive')
```

Reply:
Done! Thanks for the suggestions!